src/compile/mark/mark.ts

```diff
@@ -275,7 +275,7 @@ function stackTransform(model: UnitModel, stackFields: string[]) {
   // add stack transform to mark
   let transform: VgStackTransform = {
     type: 'stack',
-    groupby: [model.field(stack.groupbyChannel, {binSuffix: '_mid'})],
+    groupby: [model.field(stack.groupbyChannel, {binSuffix: 'mid'}) || 'undefined'],
```

Comment:
Similar to here, probably should assign group by only if this field value is not null.  

(We also care out stack transform order that output should come later, so might need to use our nested `extend` pattern.)